config/application.rb

```diff
@@ -114,13 +114,6 @@ class Application < Rails::Application
     # on the top to bottom order of initializer calls in the file.
     # Because this is easy to mess up, keep your initializers in order.
     # register plugins even before loading settings, as plugins can bring their own settings
-    initializer :register_vmdb_plugins, :before => :load_vmdb_settings do
-      Rails.application.railties.each do |railtie|
-        next unless railtie.class.name.start_with?("ManageIQ::Providers::") || railtie.try(:vmdb_plugin?)
-        Vmdb::Plugins.instance.register_vmdb_plugin(railtie)
```

Comment:
I don't think answers the question, but it's difficult.  The `:load_vmdb_settings` initializer creates the toplevel `Settings` instance and won't be reloaded in production unless something changes the actual setting values.  If the plugin is provider its own Settings, those have to be in place *before* the `:load_vmdb_settings` initializer is called.  This is why `initializer :register_vmdb_plugins, :before => :load_vmdb_settings`